Make some dependencies optional to avoid problems for consumers after 7.2.0 upgrade

Eclipse has problems after upgrading to 7.2.0 because e.g. our `org.bndtools.templating` now imports some packages because of our upgrade of Guava

See

I think the latest release of the BND tool set is kind of messed up.

There is this failure:

> 17:18:26  [ERROR] Cannot resolve dependencies of project org.eclipse.pde:org.eclipse.pde.bnd.ui:eclipse-plugin:1.2.400-SNAPSHOT
> 17:18:26  [ERROR]  with context {osgi.os=linux, osgi.ws=gtk, org.eclipse.update.install.features=true, org.eclipse.jdt.buildtime=true, osgi.arch=x86_64, org.eclipse.update.install.sources=true}
> 17:18:26  [ERROR]   Software being installed: org.eclipse.pde.bnd.ui 1.2.400.qualifier
> 17:18:26  [ERROR]   Missing requirement: org.bndtools.templating 7.2.0.202512261929 requires 'java.package; com.google.common.util.concurrent.internal 0.0.0' but it could not be found
> 17:18:26  [ERROR]   Cannot satisfy dependency: org.eclipse.pde.bnd.ui 1.2.400.qualifier depends on: java.package; org.bndtools.templating [2.0.0,3.0.0): See log for details

Looking at the new version of that bundle, I think this import should probably be `Private-Package` like the other `com.google.common` packages:

<img width="1706" height="1427" alt="image" src="https://github.com/user-attachments/assets/0f739141-2a79-4438-a57e-9013c91f17f8" />

Looking at

https://repo.maven.apache.org/maven2/org/bndtools/org.bndtools.templating/7.2.0/org.bndtools.templating-7.2.0.pom

the dependency listed as `provided`:

> <dependency>
> <groupId>com.google.guava</groupId>
> <artifactId>guava</artifactId>
> <version>33.4.8-jre</version>
> <scope>provided</scope>
> </dependency>

I don't think we want to drag guava into the Platform/SDK.

---

I don't think that's the only think that' appears messed up....

When I try to build these updates with Orbit, that build fails like this:

> [ERROR]   Missing requirement: biz.aQute.tester.junit-platform 7.2.0.202512261929 requires 'java.package; org.junit.platform.commons [1.10.1,1.13.0)' but it could not be found
> [ERROR]   Cannot satisfy dependency: org.eclipse.orbit.maven.osgi.all.feature.group 4.39.0.v20251227-1000 depends on: org.eclipse.equinox.p2.iu; biz.aQute.tester.junit-platform [7.2.0.202512261929,7.2.0.202512261929]
> [ERROR]   Cannot satisfy dependency: org.eclipse.orbit.maven.osgi.site raw:1.0.0.'SNAPSHOT'/format(n[.n=0;[.n=0;[-S]]]):1.0.0-SNAPSHOT depends on: org.eclipse.equinox.p2.iu; org.eclipse.orbit.maven.osgi.all.feature.group 0.0.0: See log for details
> [ERROR] -> [Help 1]

The current version of the `org.junit.platform.commons` package is `1.14.1`:

<img width="587" height="317" alt="image" src="https://github.com/user-attachments/assets/b0480d64-d8d8-49d0-a126-2f7346f91ba6" />

The older version of that bundle has a broader range for the dependency:

<img width="1885" height="910" alt="image" src="https://github.com/user-attachments/assets/251739a5-e5c8-42ef-b328-9ff268a4071a" />

So this too appear inappropriate to me.

---

I can't be sure there are other problems we've not noticed yet...

_Originally posted by @merks in https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/3579#issuecomment-3693797403_
            


## Tasks

Make following optional:

- com.google.errorprone.annotations
- com.google.errorprone.annotations.*


Before:

```
 com.google.common.util.concurrent.internal
 com.google.errorprone.annotations.concurrent
 com.google.errorprone.annotations
```

Goal:

```
com.google.common.util.concurrent.internal;resolution:=optional
com.google.errorprone.annotations.concurrent;resolution:=optional
com.google.errorprone.annotations;resolution:=optional
```

--- a/org.bndtools.templating/bnd.bnd
+++ b/org.bndtools.templating/bnd.bnd
@@ -17,8 +17,8 @@
 	org.antlr:ST4:jar:complete;maven-scope=provided,\
 	com.github.spullara.mustache.java:compiler;maven-scope=provided,\
 	com.google.guava;version="[33.4.8,33.4.9)";maven-scope=provided,\
-	org.jspecify.jspecify;version='[1.0.0,2.0.0)';maven-scope=provided
-
+	com.google.guava.failureaccess;version=latest;maven-scope=provided
+	
 -testpath: \
 	slf4j.api,\
 	slf4j.simple,\
@@ -32,7 +32,7 @@
 	aQute.libg.*,\
 	com.github.mustachejava.*,\
 	com.google.common.*,\
-	org.jspecify.*
+	
 	
 Bundle-ActivationPolicy: lazy
 
@@ -41,6 +41,8 @@ Import-Package: \
  com.google.appengine.*;resolution:=optional,\
  com.google.apphosting.*;resolution:=optional,\
  android.os.*;resolution:=optional,\
+ !org.jspecify.*,\
+ !com.google.errorprone.annotations.*,\
  ${eclipse.importpackage},\
  *
 

